File: app/lib/backend/schema/tag.dart
class Tag {
  final String id;
  final String uid;
  final String name;
  final String? color;
  final DateTime createdAt;
  final DateTime updatedAt;

  Tag({
    required this.id,
    required this.uid,
    required this.name,
    this.color,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Tag.fromJson(Map<String, dynamic> json) {
    return Tag(
      id: json['id'],
      uid: json['uid'],
      name: json['name'],
      color: json['color'],
      createdAt: DateTime.parse(json['created_at']).toLocal(),
      updatedAt: DateTime.parse(json['updated_at']).toLocal(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'uid': uid,
      'name': name,
      'color': color,
      'created_at': createdAt.toUtc().toIso8601String(),
      'updated_at': updatedAt.toUtc().toIso8601String(),
    };
  }
}

File: backend/models/tag.py
from datetime import datetime, timezone
from typing import Optional
from pydantic import BaseModel, Field


class Tag(BaseModel):
    id: str = Field(description="Unique identifier for the tag")
    uid: str = Field(description="User ID who owns this tag")
    name: str = Field(description="Name of the tag")
    color: Optional[str] = Field(default=None, description="Hex color code for the tag (e.g., #FF5733)")
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))


class CreateTag(BaseModel):
    name: str = Field(description="Name of the tag")
    color: Optional[str] = Field(default=None, description="Hex color code for the tag")


class UpdateTag(BaseModel):
    name: Optional[str] = None
    color: Optional[str] = None
